### ISSUE TITLE
VIBES_GEOMETRY_CHANGING_PROPERTIES

### ISSUE THREAD

**Issue by benEnsta** [ENSTABretagneRobotics/VIBES#104]
What is the goal of VIBES_GEOMETRY_CHANGING_PROPERTIES macro in vibesgraphicsitem.h

**Comment by dvinc** [comment]
Hi Benoit,
The `VIBES_GEOMETRY_CHANGING_PROPERTIES` macro generates an override of the `bool propertyChangesGeometry(const QString & key)` function, when defining _VibesGraphicsItem_ subclasses.
It has to be used in the class definition of the subclassed _VibesGraphicsItem_, in order to declare which properties should trigger a N-D to 2-D projection recomputation when modified, because they change the geometry of the item.
Basically, all properties related to geometry, like vertices positions, radius, orientation, etc. should be listed in the `VIBES_GEOMETRY_CHANGING_PROPERTIES("prop1", "prop2", ...)` macro statement.
Similarly, the `VIBES_READ_ONLY_PROPERTIES` macro enables to define some properties as read-only (in addition to "type" which is always read-only).

**Comment by benEnsta** [comment]
Hi Vincent,
Thanks for the explanations.
